### DISCUSSION TITLE
JAX and CUDA compatibility when installing JAX into WSL2

### DISCUSSION THREAD

**Discussion by ToshiyukiBandai** [jax-ml/jax#21435] [Q&A] [answered]
Hi all,

I'm installing jax.0.4.28 into Windows Linux Subsystem 2 (WSL2). I understand the support for WSL2 is experimental, but I want to share some experiences to hopefully get some remedies.

After installing jax via the commad as in the installation document
```
pip install -U "jax[cuda12]"
```
I test running JAX and got the following error
![image](https://github.com/google/jax/assets/46998790/342bc1a8-1891-424c-9bdb-c9f5b5b9ceae)

This error disappears after a trick (#18027)

```
python3 -m pip install tensorflow[and-cuda]
```

I don't know why the error disappeared because by running `nvidia-smi` I can see the version of CUDA is 12.3, not 12.5, which was required at first. 

![image](https://github.com/google/jax/assets/46998790/271ff076-605d-4413-80d2-dfce2f4db499)

Does anyone know what is going on?

**Comment by ToshiyukiBandai** [comment, marked as answer]
For record, I just needed to install the latest version of NVIDIA driver (https://docs.nvidia.com/cuda/wsl-user-guide/index.html). Make sure to install the driver into the Windows system only, not the WSL2 Linux system. Unfortunately, it seems some bug in WSL2 showing "segmentation fault" if you check the driver version by nvidia-smi (https://github.com/microsoft/WSL/issues/11277). I got the message on WSL2 on WIndows 11 but not WSL2 on Windows10.

**Comment by hawkinsp** [comment]
Yes, that is a warning saying that your CUDA driver is older than your `ptxas` version. Upgrading your driver will work.

`python3 -m pip install tensorflow[and-cuda]` works also because that will downgrade `ptxas`, since tensorflow depends on a specific older version. In general I'd recommend upgrading rather than downgrading!
